staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -460,6 +461,10 @@ func (config *DirectClientConfig) getCluster() (clientcmdapi.Cluster, error) {
 		mergedClusterInfo.CertificateAuthorityData = config.overrides.ClusterInfo.CertificateAuthorityData
 	}
 
+	if config.overrides.ClusterInfo.TLSServerName != "" {
```

Comment:
since tls server name defaults to server, I would expect setting Server and TLSServerName in the overrides to be paired. for example, if I have a kubeconfig that has this:
```
tlsServerName: https://example.com
server: https://1.2.3.4
```
and I override `--server https://foo.com`, I would expect `foo.com` to be used to verify the server unless I also override `--tls-server-name https://example.com`